src/rules/prefer-default-export.js

```diff
@@ -88,8 +103,15 @@ module.exports = {
       },
 
       'Program:exit': function () {
-        if (specifierExportCount === 1 && !hasDefaultExport && !hasStarExport && !hasTypeExport) {
-          context.report(namedExportNode, 'Prefer default export.');
+        if (target === 'single') {
+          if (specifierExportCount === 1 && !hasDefaultExport && !hasStarExport && !hasTypeExport) {
+            context.report(namedExportNode, SINGLE_EXPORT_ERROR_MESSAGE);
+          }
+          return;
+        } else if (target === 'any') {
+          if (specifierExportCount > 0 && !hasDefaultExport && !hasStarExport && !hasTypeExport) {
```

Comment:
seems like we can early return if `hasDefaultExport || hasStarExport || hasTypeExport`, for both code paths?

Reply:
yeah, that makes sense.